### ISSUE TITLE
How to choose best checkpoints?

### ISSUE THREAD

**Issue by pgsrv** [Alibaba-MIIL/ASL#46]
Hi! I'm using ASL+EMA for imbalanced multi-label problem. For ordinary softmax problems avg_checkpoints.py from timm works very good. I've tried checkpoint averaging for ASL+EMA, the results are very different, they highly depend on the selected checkpoints. Are there approaches to choose best checkpoints to average them? And should we use checkpoint averaging with EMA at all?

**Comment by mrT23** [comment]
I don't think your question relates to ASL.
you can train with ASL without EMA.
on multi-label problems, we found that EMA helps a lot (for any loss function).

so my first recommendation is to try EMA +cross-entropy on your problem, and see if results are less dependent on the selected checkpoints. my strong guess - no.

i think that  'checkpoint averaging'+'no ema' will be somewhat equivalent to EMA. with EMA, i don't think you get a big benefit from 'checkpoint averaging'.

**Comment by pgsrv** [comment]
Thank you very much!
